sdks/java/core/src/main/java/org/apache/beam/sdk/values/reflect/RecordFactory.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.values.reflect;
+
+import com.google.common.collect.ImmutableList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import net.bytebuddy.ByteBuddy;
+import org.apache.beam.sdk.values.BeamRecord;
+import org.apache.beam.sdk.values.BeamRecordType;
+import org.apache.beam.sdk.values.reflect.field.DirectAccessGetterFactory;
+import org.apache.beam.sdk.values.reflect.field.FieldValueGetter;
+import org.apache.beam.sdk.values.reflect.field.GetterMethodGetterFactory;
+
+/**
+ * Generates the code to create BeamRecordTypes and BeamRecords based on pojos.
+ *
+ * <p>Generated record types are cached in the instance of this factory.
+ *
+ * <p>Currently only 1-1 mapping is supported between pojo classes and record types.
```

Comment:
<!--thread_id:cc_154608554_t; commit:12389cee6a5061c472533ec063d041e03007687b; resolved:1-->
<!--section:context-quote-->
> **iemejia** wrote:
> Maybe add that only 'primitive' types are supported because it does not support more complex types, Lists/Sets/Maps, etc.

<!--section:body-->
This class doesn't explicitly limit what types are supported. It uses `RecordTypeFactory` to create record types. After updating `DefaultRecordTypeFactory` to use `CoderRegistry` it should allow using a lot more types, e.g. byte arrays, sets, serializables. Not sure what would be the use of it at the moment, because it's not used in SQL. `SqlRecordTypeFactory` is the one used for SQL and it does not support complex types at the moment.